src/scipp/io/hdf5.py

```diff
@@ -281,44 +277,44 @@ def _read_array_data(cls, group: h5.Group, data: Variable) -> None:
 
     @classmethod
     def write(cls, group: h5.Group, var: Variable) -> h5.Group | None:
-        if var.dtype not in cls._dtypes.values():
+        if var.bins is None and var.dtype not in cls._dtypes.values():
             # In practice this may make the file unreadable, e.g., if values
             # have unsupported dtype.
             get_logger().warning(
                 'Writing with dtype=%s not implemented, skipping.', var.dtype
             )
             return None
         _write_scipp_header(group, 'Variable')
-        dset = cls._write_data(group, var)
+        if var.bins is None:
```

Comment:
I'm not sure I understand from the changes if we would now be able or not able to read old scipp .h5 files that contain binned data...
Can you help?